src/test/java/jp/kusumotolab/kgenprog/output/PatchSerializerTest.java

```diff
@@ -0,0 +1,130 @@
+import org.junit.Before;
+import org.junit.Test;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import jp.kusumotolab.kgenprog.ga.Base;
+import jp.kusumotolab.kgenprog.ga.Fitness;
+import jp.kusumotolab.kgenprog.ga.Gene;
+import jp.kusumotolab.kgenprog.ga.HistoricalElement;
+import jp.kusumotolab.kgenprog.ga.MutationHistoricalElement;
+import jp.kusumotolab.kgenprog.ga.OriginalHistoricalElement;
+import jp.kusumotolab.kgenprog.ga.SimpleFitness;
+import jp.kusumotolab.kgenprog.ga.Variant;
+import jp.kusumotolab.kgenprog.project.GeneratedSourceCode;
+import jp.kusumotolab.kgenprog.project.ProductSourcePath;
+import jp.kusumotolab.kgenprog.project.factory.TargetProject;
+import jp.kusumotolab.kgenprog.project.factory.TargetProjectFactory;
+import jp.kusumotolab.kgenprog.project.jdt.GeneratedJDTAST;
+import jp.kusumotolab.kgenprog.project.jdt.InsertOperation;
+import jp.kusumotolab.kgenprog.project.jdt.JDTASTConstruction;
+import jp.kusumotolab.kgenprog.project.jdt.JDTASTLocation;
+import jp.kusumotolab.kgenprog.project.test.EmptyTestResults;
+import jp.kusumotolab.kgenprog.testutil.JsonKeyAlias;
+import jp.kusumotolab.kgenprog.testutil.TestUtil;
+
+public class PatchSerializerTest {
+
+  private Gson gson;
+  private JDTASTConstruction astConstruction = new JDTASTConstruction();
+  private PatchGenerator patchGenerator = new PatchGenerator();
+
+  @Before
+  public void setup() {
+    gson = new GsonBuilder().registerTypeAdapter(Patch.class, new PatchSerializer())
+        .registerTypeHierarchyAdapter(FileDiff.class, new FileDiffSerializer())
+        .create();
+  }
+
+  private Variant createVariant(final Fitness fitness,
+      final TargetProject targetProject) {
+
+    return new Variant(0, 0, new Gene(Collections.emptyList()),
+        astConstruction.constructAST(targetProject),
+        EmptyTestResults.instance, fitness, Collections.emptyList(),
+        new OriginalHistoricalElement());
+  }
+
+  private Variant createVariant(final long id, final int generationNumber, final Fitness fitness,
+      final GeneratedSourceCode code, final HistoricalElement historicalElement) {
+    return new Variant(id, generationNumber, new Gene(Collections.emptyList()), code,
+        EmptyTestResults.instance, fitness, Collections.emptyList(), historicalElement);
+  }
+
+  /**
+   * パッチが下の形式でシリアライズされるかテストする
+   *
+   * [{ "fileName":"ファイル名(fqn)", "diff":"元のプログラムとの差分" }]
+   */
+  @Test
+  public void testPatch() {
+    // 初期Variant
+    final Path rootPath = Paths.get("example/CloseToZero01");
+    TargetProject project = TargetProjectFactory.create(rootPath);
+    final Variant initialVariant = createVariant(new SimpleFitness(0.0d), project);
+
+    // 差分を作るために適当な位置にコードを挿入する
+    final GeneratedSourceCode originalSourceCode = TestUtil.createGeneratedSourceCode(project);
+    final GeneratedJDTAST ast = (GeneratedJDTAST) originalSourceCode.getProductAsts()
```

Comment:
警告が出るのでジェネリクスを書きましょう
```suggestion
    final GeneratedJDTAST<ProductSourcePath> ast = (GeneratedJDTAST<ProductSourcePath>) originalSourceCode.getProductAsts()
```